File: cPP_java_py_js-DSA/data-structure/set/Set.js
function printSet(s) {
  console.log("Set values ");

  for (let val of s) {
    console.log(val + " ");
  }
}
let s = new Set();
console.log(s);

// 1. insert(value) operation = add(value) operation in set
s.add(2);
s.add(2); // duplicate values will not be inserted into set
s.add(4);
s.add(10);
printSet(s);

//2. erase(value) operation = delete(value) operation in set
s.delete(4);
printSet(s);

//3. find() method =  has() method in set
console.log(s.has(4));
console.log(s.has(2));

//4.check an element exist or not
if (s.has(4)) {
  console.log("Element exits");
} else {
  console.log("Not exists ");
}

//4.clear() method = clear() method in set also : delete all the values from set
// s.clear();

//5. size () method  = size property in set  = to get the size of set
console.log(s.size);

//6. iterating over set value : using forOf loop
for (let val of s) {
  console.log("value", val);
}

File: cPP_java_py_js-DSA/data-structure/set/set.cpp
#include<bits/stdc++.h>
#include<set>
using namespace std ; 

void printSet(unordered_set<int>s ){
    cout<<"Set value : "; 
    for(auto it: s ){
        cout<<it<<", ";
    }
    cout<<endl;

}
int main(){

    // set creation 
    unordered_set<int>s;


    // 1. insert Operation : for adding vauees in set
    s.insert(2);
    s.insert(4);
    s.insert(6);
    s.insert(8);
    printSet(s);

    // 2. erase operation : for remove values from set via value
    s.erase(4);
    printSet(s);

    // find method : to find particular value in a set :-> it will give a iterator
    (s.find(2)); 
    printSet(s);

    //4.check if an element exists in set or not 
    int element =2; 
    if(s.find(element)!= s.end()){
        cout<<element<<"Exists in array "<<endl;
    }

    // 5. size method : to get the no of elements present in set 
    cout<<s.size()<<endl;

    // 6. clear method : to delete all the elements of set 
    // s.clear(); 
    printSet(s);

    //7 Ierating over set for(auto val) loop: to get the each value of set and perfom some operation
    for(auto it: s ){
        cout<<it<<" ";
    }
    cout<<endl;

    //8. count method : to find number of elements in set 
    cout<<s.count(6);

   

    cout<<"\nConcept of Set and its available operations \n";
    return 0;
}